src/contrib/cluster/Akka.Cluster.Tools/Singleton/OldestChangedBuffer.cs

```diff
@@ -153,10 +153,15 @@ private bool MatchingRole(Member member)
         private void HandleInitial(ClusterEvent.CurrentClusterState state)
         {
             _membersByAge = state.Members
-                .Where(m => (m.Status == MemberStatus.Up || m.Status == MemberStatus.Leaving) && MatchingRole(m))
```

Comment:
Makes sense - wouldn't want to pin a `ClusterSingleton` on a node that's leaving.